### PR TITLE
feat: add grid overlay to responsive matrix (#59)

### LINKED ISSUE #59: [FEAT]: First draft of the responsive matrix cells
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mj3cheun**
# Description

Add a procedurally generated grid overlay to the responsive matrix component.

Resolves #59

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				fragColor,
+				rowGrid,
+				1. - smoothstep(
+					uGridRowDisplayLim - uGridRowDisplayTransition,
+					uGridRowDisplayLim + uGridRowDisplayTransition,
+					uViewportWorldHeight / uWorldHeight * uNumRow
+				)
+			)
+		);
+	}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:82]: This `fragColor. max(...)` is hard to read and needs some comment love.

And why do we need to do interpolation?

Comment (mj3cheun) [reply packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:82]: the interpolation is for anti-aliasing, the aliasing is really really bad without it

File: packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+uniform float uViewportWorldHeight;
+uniform bool uGridDisplayBorder;
+uniform float uGridColDisplayLim;
+uniform float uGridRowDisplayLim;
+uniform float uGridColDisplayTransition;
+uniform float uGridRowDisplayTransition;
+out vec4 fragColor;
+void main() {
+	float microColLen = float(textureSize(uMicroCol, 0).x);
+	float microRowLen = float(textureSize(uMicroRow, 0).x);
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:26]: Maybe
- microColCellSize, and
- microRowCellSize 

Makes reading a little easier to read?  they kind of look like they are referring to the length of an array data structure.

Comment (mj3cheun) [reply packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:26]: they are referring to the length of the array data structure

File: packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts (hunk 1 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:105]: @mj3cheun I am referring to lines 95 to 103

Comment (mj3cheun) [reply packages/client/hmi-client/src/components/responsive-matrix/matrix-grid.fs.glsl.ts:105]: sorry, when i pushed more changes it changed the line numbers. yup thats what im talking about, lines 95 to 103

**Review by mwdchang — APPROVED** [review]

**Pull request merged**